core/completion.py

```diff
@@ -110,17 +110,29 @@ def __init__(self):
 ST_Other = 2
 
 # State machine definition.
-_TRANSITIONS = {
-    # (state, char) -> (new state, emit span)
-    (ST_Begin, CH_Break): (ST_Break, False),
-    (ST_Begin, CH_Other): (ST_Other, False),
+# (state, char) -> (new state, emit span)
```

Comment:
I would add a comment: This looks weird in Python because it used to be a dict, but C++ compilers should be very good at turning this function into a lookup table (I tried this once, e.g. with Godbolt)